### ISSUE TITLE
Different display of cli in PowerShell and Command Prompt

### ISSUE THREAD

**Issue by bonskijr** [dbcli/mssql-cli#111]
mssql-cli displays differently when being used inPowershell console. In command prompt it displays as it should but in Powershell it display filler "rectangle" for spaces.

Poweshell

![image](https://user-images.githubusercontent.com/60311/33982923-54263c96-e0ed-11e7-9a51-df29c41e73d7.png)

Command Prompt

![image](https://user-images.githubusercontent.com/60311/33982954-61518556-e0ed-11e7-9802-d5d5ad1d48b9.png)

mssql-cli v 0.7 .0
OS: Windows 10.0.15063
PowerShell : 5.1.15063.726

**Comment by abhisheksinha89** [comment]
Hi, I am not able to repro this on Win10 Powershell 5.1 
Can you tell me what codepage you are using for the powershell terminal?

**Comment by bonskijr** [comment]
The original codepage was 437 (US)
![image](https://user-images.githubusercontent.com/60311/34187364-fa882d2e-e56b-11e7-941b-42cd056fdfd6.png)

Then after running the cli changed to 65001 (UTF-8)

![image](https://user-images.githubusercontent.com/60311/34187350-e5072fb8-e56b-11e7-931c-346c7e99836d.png)

But as you can see both the issues, at least for me

**Comment by abhisheksinha89** [comment]
Thanks I'll need to investigate this. I can't seem to repro this on 2 of our machines using version 0.8.0.
I noticed that you still have version 0.7.0.
Try doing a pip install --upgrade mssql-cli 

Eventhough I think we didn't make any major changes just try upgrading and see if the problem persists. I'll try to get a repro so I can investigate this on my end.

**Comment by bonskijr** [comment]
Hi @abhisheksinha89 ,

I tried upgrading but issue still persist, but thanks for looking into this.

![image](https://user-images.githubusercontent.com/60311/34187771-fd956aac-e56d-11e7-992d-bf88e5de4faa.png)

**Comment by snewfie** [comment]
I having the issue also and using 65001(UTF-8) code page is there a fix in sight.
using windows 10 and powershell shell also occurs on cmd shell.

![image](https://user-images.githubusercontent.com/4139281/34566152-90c98cea-f12b-11e7-889f-7a3a25cef427.png)
![image](https://user-images.githubusercontent.com/4139281/34566194-b5df94de-f12b-11e7-9861-adee4f8ff629.png)

But if I use the cmd shell in legacy mode works fine 

![image](https://user-images.githubusercontent.com/4139281/34566262-e8d428d2-f12b-11e7-81d9-38ae9e8b3278.png)

![image](https://user-images.githubusercontent.com/4139281/34566272-ef0816aa-f12b-11e7-937a-325eb0f43471.png)

 hope this helps

**Comment by MrMeemus** [comment]
I was able to repro this issue even with a simple prompt from prompt tool kit. Could potentially be a renderer issue. Will keep digging and update here.

**Comment by MrMeemus** [comment]
@bonskijr @snewfie , I wasn't able to pinpoint a exact issue from the source. It looks like it could be a font issue with decoding the transparent characters within the prompt toolkit. I was able to fix this issue by switching my font (Lucida Console) for Powershell to some other font (Raster fonts) which removed the rectangular characters. I also verified switching the new font back to the original font did not return the rectangles. 

It could have been a font mismatch in the mapping where the OS wasn't font substituting properly.

TL;DR

- Change the font to some other font and then back to the original.

Either way, let me know if this resolved your issue. Thanks!

**Comment by jonathanslenders** [comment]
This looks like a prompt_toolkit issue indeed. I can reproduce it with the example prompt applications.

Which Python version are you using? I don't remember that we were changing the code page, but I think Python 3.6 does. In any case, I think we can probably fix it.

Also notice that prompt_toolkit 2.0 (non yet released), can render using vt100 escape sequences on Windows 10, which is also more efficient.

**Comment by bonskijr** [comment]
hi @jonathanslenders , running Python 2.7.1 so I can tinker with Sql op studio source(requirement).

But I'm not sure what I installed or applied but it looks ok to me now

![image](https://user-images.githubusercontent.com/60311/35615216-70d6905e-06ac-11e8-948c-1f6ad7e71c0f.png)

**Comment by MrMeemus** [comment]
@bonskijr , did you happen to change your font since the first time you reported the issue?

**Comment by snewfie** [comment]
changing the font fixes the issue wihout needing to use legacy mode

used consolas

thx for all the input guys

**Comment by snewfie** [comment]
![image](https://user-images.githubusercontent.com/4139281/35640328-dc93831e-068a-11e8-9dd5-7198dda2e913.png)

**Comment by bonskijr** [comment]
@MrMeemus no changes in font, default was Consolas ( see first screenshots ). 

I think its probably related to codepage, at first cmd/ps, my codepage s 437 and upon exit from mssql-cli changes to 65001

![image](https://user-images.githubusercontent.com/60311/35654606-d947684e-0728-11e8-9709-508afef2ac83.png)

note: not sure why this changed to v.0.9 of mssq-cli, probably because I also have the source downloaded?

**Comment by MrMeemus** [comment]
Closing this issue with the troubleshooting steps in place merged above.

**Comment by pensivebrian** [comment]
Let's leave this open the issue still exists, we just have a workaround.
